openfisca_core/entities.py

```diff
@@ -38,20 +143,20 @@ def to_json(cls):
     # Calculations
 
     def check_variable_defined_for_entity(self, variable_name):
-        if not (self.simulation.get_variable_entity(variable_name) == self):
-            variable_entity = self.simulation.get_variable_entity(variable_name)
-            raise Exception(
+        variable_entity = self.simulation.tax_benefit_system.get_column(variable_name, check_existence = True).entity
```

Comment:
We use `check_variable_defined_for_entity` when we are building the entity.
At this stage, `simulation.entities` is not yet defined, so  `self.simulation.get_variable_entit` does not work.